### PR TITLE
feat: new question card component

### PR DESCRIPTION
**Pull request by JennaXiao3**
## Notion ticket link
Update Question Card component (https://www.notion.so/uwblueprintexecs/Update-Question-Card-component-b33ea6f8d0d5411f96453d296abc1194)

## Implementation description
* Added the necessary icons
* Modified QuestionCard props (removed image and text; added tags, questions, points, questionNum)
* Created a QuestionTag component to display the tags accordingly 

## Steps to test
1. Go to http://localhost:3000/component-library
2. Scroll down to the question card component
3. Ensure that ellipsis show when text overflows
4. Ensure everything looks correct according to the Figma 

## What should reviewers focus on?
* if the result matches the design
* ensure that the props are correct

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/themes/typography.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { getByPlaceholderText } from "@testing-library/react";
```
Comment (carissa-tang) [inline frontend/src/themes/typography.ts]: what is this used for?

File: frontend/src/components/common/QuestionTag.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+export enum QuestionType {
+  MULTIPLE_CHOICE = "Multiple Choice",
+  SHORT_ANSWER = "Short Answer",
+  MULTI_SELECT = "Multi-select",
+}
+
+type QuestionTypeProps = {
+  type: QuestionType;
+  frequency: number;
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionTag.tsx]: nit: can you rename this field from frequency -> count?

File: frontend/src/components/common/QuestionTag.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { Tag, Text, TagLeftIcon, Box } from "@chakra-ui/react";
+import {
+  MultiSelectTagIcon,
+  MultipleChoiceTagIcon,
+  ShortAnswerTagIcon,
+} from "../../assets/icons"; // icons might be duplicated
+
+export enum QuestionType {
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionTag.tsx]: can you move this to `src/types/QuestionTypes.ts`?

File: frontend/src/components/common/QuestionTag.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    <Tag
+      color={color}
+      bgColor={bgColor}
+      padding="10px 14px"
+      whiteSpace="nowrap"
+      minWidth="fitContent"
+      borderRadius="full"
+      fontSize="16px"
+    >
+      <TagLeftIcon boxSize="16px" as={icon} />
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionTag.tsx]: nit: slightly bigger icon?

File: frontend/src/components/common/QuestionTag.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { Tag, Text, TagLeftIcon, Box } from "@chakra-ui/react";
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionTag.tsx]:
```suggestion
import { Tag, Text, TagLeftIcon } from "@chakra-ui/react";
```

File: backend/models/test.model.ts [hunk 6]
```diff
@@ -6,6 +6,8 @@ import mongoose, { Schema, Document } from "mongoose";
 export enum QuestionType {
   MULTIPLE_CHOICE,
   NUMERIC_ANSWER,
+  SHORT_ANSWER,
```
Comment (carissa-tang) [inline backend/models/test.model.ts]: can you omit this change for now? i will add it in my pr!

File: frontend/src/components/common/QuestionCard.tsx [hunk 7]
```diff
@@ -1,28 +1,40 @@
-import React from "react";
+import * as React from "react";
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionCard.tsx:1]: is this change needed?

File: frontend/src/components/common/QuestionTag.tsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+import {
+  MultiSelectTagIcon,
+  MultipleChoiceTagIcon,
+  ShortAnswerTagIcon,
+} from "../../assets/icons"; // icons might be duplicated
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionTag.tsx]: what does your comment mean?

File: frontend/src/components/common/QuestionCard.tsx [hunk 9]
```diff
@@ -1,28 +1,40 @@
...
   Container,
   Flex,
-  Image,
   Stack,
-  Tag,
   Text,
+  Menu,
+  MenuButton,
+  IconButton,
+  Button,
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionCard.tsx]:
```suggestion
```

File: frontend/src/components/common/QuestionCard.tsx [hunk 10]
```diff
@@ -1,28 +1,40 @@
...
 } from "@chakra-ui/react";
+import { QuestionType, QuestionTag } from "./QuestionTag";
+import {
+  EditOutlineIcon,
+  DeleteOutlineIcon,
+  HamburgerMenuIcon,
+} from "../../assets/icons";
 
 type QuestionCardProps = {
   date: Date;
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionCard.tsx]: you can remove date and title i believe

File: frontend/src/components/common/QuestionCard.tsx [hunk 11]
```diff
@@ -1,28 +1,40 @@
...
   date,
   title,
-  image,
-  text,
   tags,
+  questionNum,
+  points,
+  questions,
 }: QuestionCardProps): React.ReactElement => {
   const formatDate = (d: Date) =>
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionCard.tsx]: this function can be removed

File: frontend/src/components/common/QuestionCard.tsx [hunk 12]
```diff
@@ -32,67 +44,80 @@ const QuestionCard = ({
...
+              flexGrow="1"
+              marginBottom="20px"
+              textStyle="subtitle1"
+              align="left"
+              color="grey.400"
+            >
+              Question {questionNum}
+            </Text>
+            <Box color="blue.300" fontSize="24px" mr="4">
+              <EditOutlineIcon />
```
Comment (carissa-tang) [inline frontend/src/components/common/QuestionCard.tsx:73]: can you add `cursor: pointer;` to the two new icons

**Review by carissa-tang — CHANGES_REQUESTED** [review]
great work !! left some comments

File: frontend/src/themes/typography.ts (hunk 1 above)
Comment (JennaXiao3) [reply frontend/src/themes/typography.ts]: I'm pretty sure I didn't add that...I can remove it if you would like

Comment (JennaXiao3) [reply frontend/src/themes/typography.ts]: I have removed it

File: frontend/src/App.tsx [hunk 13]
```diff
@@ -99,6 +100,12 @@ const App = (): React.ReactElement => {
                   path={Routes.EMAIL_ACTION}
                   component={EmailActionHandler}
                 />
+                <PrivateRoute
```
Comment (carissa-tang) [inline frontend/src/App.tsx:104]: can you add a quick comment?
```suggestion
// TODO: Remove route for production
                <PrivateRoute
```

**Review by carissa-tang — APPROVED** [review]
just quick comment, otherwise LGTM !! 🎉 woohoo
